lib/sharing/ConsumeSharedPlugin.js

```diff
@@ -252,6 +256,62 @@ class ConsumeSharedPlugin {
 								requiredVersion
 							}
 						);
+						// Check if exclusion criteria is provided
+						if (config.exclusionCriteria && config.exclusionCriteria.version) {
+							// Without a resolved import, we cannot proceed with exclusion check
+							if (!importResolved) {
+								return consumedModule;
+							}
+							// If a fallback version is provided by the user, use that to check
+							if (config.exclusionCriteria.fallbackVersion) {
+								if (
+									satisfy(
+										parseRange(config.exclusionCriteria.version),
+										config.exclusionCriteria.fallbackVersion
+									)
+								) {
+									return undefined;
+								}
+								return consumedModule;
+							}
+							return new Promise(resolve => {
+								// Get the closest package.json file corresponding to the resolved import.
+								// This version will be used for checking whether the dependency must be
+								// excluded from consumption (in a share context). If such a version does
+								// not exist in the package.json file, then we use the "version" property
+								// set by users in the share config, which is passed here as fallbackVersion
+								// a.k.a the version used to register the package in the scope.
+								getDescriptionFile(
```

Comment:
Find the closest `package.json` for the resolved import. This gives us the version of the package/module that corresponds to the request. Note that there are shared modules that do not have a corresponding `package.json` file, and for these, we do not exclude them from sharing *unless* the user supplies a `version` property. If that is the case, the check in line 269 will activate